### PR TITLE
upgrade eslint from 3.12.0 to 6.0.0

### PR DESCRIPTION
**Pull request by abehrman89**
## Security Alert
- resolves `js-yaml` [high-security alert](https://github.com/policygenius/athenaeum/network/alert/yarn.lock/js-yaml/open)
- all `js-yaml` now resolves to `3.13.1`

## `eslint`
- upgrade `eslint` from `3.12.0` to `6.0.0`
- upgrade `es-loader` from `1.7.1` to `3.0.0`
- upgrade `eslint-plugin-import` from `2.2.0` to `2.19.1`
- upgrade `eslint-plugin-jsx-a11y` from `2.2.3` to `6.2.3`
- upgrade `eslint-plugin-react` from `6.8.0` to `7.17.0`

- `.eslintrc` now extends `policygenius` instead of `airbnb`
- remove `eslint-config-airbnb` and `eslint-config-airbnb-base`
- add `eslint-config-policygenius`

### CODE REVIEW

File: src/atoms/Element/index.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ import PropTypes from 'prop-types';
 import classnames from 'classnames';
 
 /* eslint-disable import/first */
-import styles from './element.module.scss';
 import colors from 'atoms/Color/colors.scss';
 import spacing from 'atoms/Spacer/spacer.scss';
+import styles from './element.module.scss';
```
Comment (jmcolella) [inline src/atoms/Element/index.js]: can you double check that moving this didn't break anything? line 5 is makes it seem like the import order was done on purpose

File: src/atoms/Icon/Readme.md [hunk 2]
```diff
@@ -107,7 +107,7 @@ _Note: this list isn't automatically kept in sync._
 To sync, run `bin/refresh_svg_list`.
 You will need to have access to our gutils
 ```jsx
-    const icons = require('./constants').icons;
+    const { icons } = require('./constants');
```
Comment (jmcolella) [inline src/atoms/Icon/Readme.md]: double check this still works as expected

File: src/atoms/Img/index.js [hunk 3]
```diff
@@ -23,14 +23,61 @@ const createName = (str) => {
   return filename ? filename[0].replace(/(\W+|_)/g, ' ').trim() : 'PolicyGenius';
 };
 
+const isPicture = (props) => {
```
Comment (jmcolella) [inline src/atoms/Img/index.js]: what was the linter error that led to this refactor?

File: src/atoms/Layout/utils.js [hunk 4]
```diff
@@ -8,15 +8,15 @@ export function processChild(child, layoutProps) {
   // TODO: Find a better way to type check here.
   // I believe the first now fails because of Preact
   if (
-    child.type === Col ||
-    get(child, 'nodeName.prototype.displayName') === 'Col' ||
-    get(child, 'type.prototype.displayName') === 'Col' ||
-    get(child, 'type.name') === 'Col' ||
-    get(child, 'type.rclName') === 'Col' ||
+    child.type === Col
+    || get(child, 'nodeName.prototype.displayName') === 'Col'
```
Comment (jmcolella) [inline src/atoms/Layout/utils.js]: what was the linter error that caused this refactor? the previous style makes more sense IMO

File: src/atoms/Tooltip/index.js [hunk 5]
```diff
@@ -69,27 +74,31 @@ class Tooltip extends React.Component {
...
-            text ||
-              <Icon
-                icon='tooltip'
-                className={styles['tooltip']}
-                inline={inline}
-                height={`${tooltipIconSize}px`}
-                width={`${tooltipIconSize}px`}
-              />
+            text
+              || (
```
Comment (jmcolella) [inline src/atoms/Tooltip/index.js]: yeah all of these `||` or `&&` updates are a bit annoying

File: src/molecules/Filter/index.js [hunk 6]
```diff
@@ -22,17 +22,19 @@ function Filter(props) {
...
-      }
+        /* eslint-enable jsx-a11y/label-has-for */
+      )}
 
       <div className={styles['select-wrapper']}>
         <select
           className={styles['select']}
           id={forProp}
-          onChange={onChange}
+          onBlur={onChange}
```
Comment (jmcolella) [inline src/molecules/Filter/index.js]: why was this updated?

File: src/molecules/asides/ImageAside/index.js [hunk 7]
```diff
@@ -44,7 +44,7 @@ function ImageAside( props ) {
           className={styles['image']}
           src={image}
           data-src={dataSrc}
-          role='presentation'
+          alt=''
```
Comment (jmcolella) [inline src/molecules/asides/ImageAside/index.js]: why did you remove `role`?

File: src/molecules/formfields/SelectField/index.js [hunk 8]
```diff
@@ -27,9 +27,25 @@ const renderAdditionalInfo = (func) => {
 };
 
 class SelectField extends React.Component {
-  render() {
+  getClasses = () => {
```
Comment (jmcolella) [inline src/molecules/formfields/SelectField/index.js]: what rule caused this refactor?

File: src/molecules/formfields/TextField/index.js [hunk 9]
```diff
@@ -13,16 +13,55 @@ import { renderTooltip } from 'utils/fieldUtils';
 import styles from './text_field.module.scss';
 
 class TextField extends React.Component {
-  render() {
+  getClasses = () => {
```
Comment (jmcolella) [inline src/molecules/formfields/TextField/index.js]: this seems like a pattern, want to know the rule for all of these changes

File: src/molecules/formfields/ToggleField/Readme.md [hunk 10]
```diff
@@ -1,4 +1,4 @@
-```jsx
+''`jsx
```
Comment (jmcolella) [inline src/molecules/formfields/ToggleField/Readme.md]: this shouldn't need to be updated. what was the rule?

File: src/molecules/lists/ArticleList/index.js [hunk 11]
```diff
@@ -17,15 +17,15 @@ function ArticleList( props ) {
...
+    <ul className={styles[type]}>
+      {data.map((item) => (
         <li className={styles['listItem']} key={item[itemProps.id]}>
           <a className={styles['link']} href={item[itemProps.link]}>
             <figure className={styles['articleWrapper']}>
               <img
                 className={styles['articleImg']}
                 src={item[itemProps.imageSrc]}
-                role='presentation'
+                alt=''
```
Comment (jmcolella) [inline src/molecules/lists/ArticleList/index.js]: why was `role` removed?

File: src/organisms/Accordion/AccordionPanel/__tests__/accordion_panel.spec.js [hunk 12]
```diff
@@ -1,8 +1,9 @@
 import React from 'react';
-import { mount } from 'enzyme';
+
+// import sinon from 'sinon';
```
Comment (jmcolella) [inline src/organisms/Accordion/AccordionPanel/__tests__/accordion_panel.spec.js]: why was this added?

File: src/organisms/Accordion/__tests__/accordion.spec.js [hunk 13]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
-import { mount } from 'enzyme';
+
+// import sinon from 'sinon';
```
Comment (jmcolella) [inline src/organisms/Accordion/__tests__/accordion.spec.js]: why added?

File: src/organisms/cards/CompanyCard/index.js [hunk 14]
```diff
@@ -46,7 +47,7 @@ const CompanyCard = (props) => {
       className={classes}
     >
       <div>
-        <img className={styles.image} src={imageAttr.src} role='presentation' />
+        <img className={styles.image} src={imageAttr.src} alt='' />
```
Comment (jmcolella) [inline src/organisms/cards/CompanyCard/index.js]: why `role` removed?

File: src/organisms/cards/PartnerCard/__tests__/partner_card.spec.js [hunk 15]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
-import { mount } from 'enzyme';
+
+// import sinon from 'sinon';
```
Comment (jmcolella) [inline src/organisms/cards/PartnerCard/__tests__/partner_card.spec.js]: why added?

File: src/organisms/cards/RadioCard/Readme.md [hunk 16]
```diff
@@ -8,7 +8,7 @@ No Image:
         super(props);
 
         this.state = {
-          radioValue: '',
+          value: '',
```
Comment (jmcolella) [inline src/organisms/cards/RadioCard/Readme.md]: why was this key changed?

File: src/organisms/cards/SimpleFeaturedPolicyCard/index.js [hunk 17]
```diff
@@ -33,53 +33,58 @@ function SimpleFeaturedPolicyCard(props) {
...
+              </div>
+            )
         }
 
-
         {
           onContinue &&
-            <React.Fragment>
-              <Spacer size={36} />
+            [
```
Comment (jmcolella) [inline src/organisms/cards/SimpleFeaturedPolicyCard/index.js]: what rule removed `React.Fragment`?

File: src/organisms/tables/ComparisonTable/index.js [hunk 18]
```diff
@@ -42,7 +42,7 @@ class TableHeader extends React.Component {
     const childrenArray = React.Children.toArray(children);
 
     if (showMissing && tableItems === 1 && childrenArray.length === 1) {
-      return childrenArray.concat([ <Col /> ]);
+      return childrenArray.concat([ <Col key={1} /> ]);
```
Comment (jmcolella) [inline src/organisms/tables/ComparisonTable/index.js]: i wouldn't hardcode this `key`. I would set it to `childrenArray.length`

File: src/organisms/tables/ComparisonTable/index.js [hunk 19]
```diff
@@ -215,7 +213,7 @@ class TableRow extends React.Component {
     const { tableItems, showMissing } = this.props;
 
     if (showMissing && tableItems === 1 && this.childrenArray.length === 2) {
-      return this.childrenArray.concat([ <Col /> ]);
+      return this.childrenArray.concat([ <Col key={1} /> ]);
```
Comment (jmcolella) [inline src/organisms/tables/ComparisonTable/index.js]: see above

File: src/utils/fieldUtils/renderTooltip.js [hunk 20]
```diff
@@ -10,7 +11,12 @@ export default function renderTooltip(tooltip, className, iconClassName) {
         right: true,
         ...overrides
       }),
-      icon: <Icon icon='tooltip' className={iconClassName} />
+      icon:
+  <Icon
```
Comment (jmcolella) [inline src/utils/fieldUtils/renderTooltip.js]: what rule led to this?

File: src/organisms/Accordion/AccordionPanel/__tests__/accordion_panel.spec.js (hunk 12 above)
Comment (abehrman89) [reply src/organisms/Accordion/AccordionPanel/__tests__/accordion_panel.spec.js]: Fixed this - accidentally added it back in when I rebased.

File: src/organisms/Accordion/__tests__/accordion.spec.js (hunk 13 above)
Comment (abehrman89) [reply src/organisms/Accordion/__tests__/accordion.spec.js]: (Same as above)
Fixed this - accidentally added it back in when I rebased.

File: src/organisms/cards/PartnerCard/__tests__/partner_card.spec.js (hunk 15 above)
Comment (abehrman89) [reply src/organisms/cards/PartnerCard/__tests__/partner_card.spec.js]: (Same as above)
Fixed this - accidentally added it back in when I rebased.

File: src/atoms/Element/index.js (hunk 1 above)
Comment (abehrman89) [reply src/atoms/Element/index.js]: I don't even remember changing this or why - I changed it back.

File: src/molecules/formfields/ToggleField/Readme.md (hunk 10 above)
Comment (abehrman89) [reply src/molecules/formfields/ToggleField/Readme.md]: I think this was changed by the linter at some point - it makes no sense so I changed it back.

File: src/organisms/tables/ComparisonTable/index.js (hunk 18 above)
Comment (abehrman89) [reply src/organisms/tables/ComparisonTable/index.js]: done

File: src/organisms/tables/ComparisonTable/index.js (hunk 19 above)
Comment (abehrman89) [reply src/organisms/tables/ComparisonTable/index.js]: done

File: src/molecules/Filter/index.js (hunk 6 above)
Comment (abehrman89) [reply src/molecules/Filter/index.js]: `onBlur must be used instead of onchange, unless absolutely necessary and it causes no negative consequences for keyboard only or screen reader users  jsx-a11y/no-onchange`

File: src/molecules/asides/ImageAside/index.js (hunk 7 above)
Comment (abehrman89) [reply src/molecules/asides/ImageAside/index.js]: the lint error was
```
 Prefer alt="" over a presentational role. First rule of aria is to not use aria if it 
can be achieved via native HTML  
jsx-a11y/alt-text
```

File: src/molecules/lists/ArticleList/index.js (hunk 11 above)
Comment (abehrman89) [reply src/molecules/lists/ArticleList/index.js]: (same as above)

the lint error was

```
Prefer alt="" over a presentational role. First rule of aria is to 
not use aria if it can be achieved via native HTML  
jsx-a11y/alt-text
```
so I replaced `role='presentation` with `alt=''`. But I just checked and it works if I keep both so I can add it back in if you'd prefer that.

File: src/organisms/cards/CompanyCard/index.js (hunk 14 above)
Comment (abehrman89) [reply src/organisms/cards/CompanyCard/index.js]: (same as above)

the lint error was

```
Prefer alt="" over a presentational role. First rule of aria is to 
not use aria if it can be achieved via native HTML  
jsx-a11y/alt-text
```
so I replaced `role='presentation` with `alt=''`. But I just checked and it works if I keep both so I can add it back in if you'd prefer that.

File: src/organisms/cards/SimpleFeaturedPolicyCard/index.js (hunk 17 above)
Comment (abehrman89) [reply src/organisms/cards/SimpleFeaturedPolicyCard/index.js]: I think this was a linter change? There's no error thrown when I add it back in so there's no reason I would have removed it.

File: src/utils/fieldUtils/renderTooltip.js (hunk 20 above)
Comment (abehrman89) [reply src/utils/fieldUtils/renderTooltip.js]: No idea - weird linter change. I've changed it back.

File: src/atoms/Img/index.js (hunk 3 above)
Comment (abehrman89) [reply src/atoms/Img/index.js]: `complexity` - I had to refactor some of the code out of the `Img` function because it was too complex

File: src/molecules/formfields/SelectField/index.js (hunk 8 above)
Comment (abehrman89) [reply src/molecules/formfields/SelectField/index.js]: `complexity` - I had to refactor some of the code out of the `render` function because it was too complex

File: src/molecules/formfields/TextField/index.js (hunk 9 above)
Comment (abehrman89) [reply src/molecules/formfields/TextField/index.js]: `complexity` - I had to refactor some of the code out of the `render` function because it was too complex

File: src/organisms/cards/RadioCard/Readme.md (hunk 16 above)
Comment (abehrman89) [reply src/organisms/cards/RadioCard/Readme.md]:
```
Unused state field: 'radioValue' 
react/no-unused-state
```

the `state` was declared as `radioValue`, but everywhere else in the file it was referred to as `value`

File: src/atoms/Icon/Readme.md (hunk 2 above)
Comment (abehrman89) [reply src/atoms/Icon/Readme.md]: Yes - all the `icons` still display correctly when I run it locally

File: src/atoms/Layout/utils.js (hunk 4 above)
Comment (abehrman89) [reply src/atoms/Layout/utils.js]: This was done automatically by the linter at some point while we were trying to update it. I hate it so I've changed it back to the way it was and it's working fine.

File: src/atoms/Tooltip/index.js (hunk 5 above)
Comment (abehrman89) [reply src/atoms/Tooltip/index.js]: Agreed. I've changed them all back to the way they were.

File: src/molecules/formfields/ShortDateField/__tests__/short_date_field.spec.js [hunk 21]
```diff
@@ -7,14 +7,15 @@ import ShortDateField from 'molecules/formfields/ShortDateField';
 
 describe('<ShortDateField />', () => {
   let props;
+
```
Comment (magcurt8) [inline src/molecules/formfields/ShortDateField/__tests__/short_date_field.spec.js]: extra line again

File: src/molecules/formfields/SliderField/Readme.md [hunk 22]
```diff
@@ -18,13 +18,15 @@
...
+        const { value } = this.state;
+
         return (
           <SliderField
             sliderValues={
               [ 1000, 5000, 25000, 500000 ]
             }
             input={{
-              value: this.state.value,
+              value,
```
Comment (magcurt8) [inline src/molecules/formfields/SliderField/Readme.md]: I'm not sure this is right, since the previous code was a prop assignment and this is just a value now, you could do `const { currentValue } = this.state;` and then `value: currentValue`

Comment (magcurt8) [reply src/molecules/formfields/SliderField/Readme.md]: there are a few other times this refactor happened as well, I just noticed it

File: src/molecules/formfields/SliderField/index.js [hunk 23]
```diff
@@ -28,17 +30,17 @@ class SliderField extends Component {
...
 
       return input.onChange(sliderAmount);
     });
   }
 
   findClosestAmount(values, value) {
-    return values.reduce((prev, curr) =>
-      Math.abs(curr - value) < Math.abs(prev - value) ? curr : prev
-    );
+    return values.reduce((prev, curr) => Math.abs(curr - value) < Math.abs(prev - value) ? curr : prev);
```
Comment (magcurt8) [inline src/molecules/formfields/SliderField/index.js]: if you add braces to this function you can keep it as two lines, the eslint `arrow-body-style` rule enforces this

File: src/molecules/formfields/TextAreaField/index.js [hunk 24]
```diff
@@ -33,7 +33,9 @@ function TextAreaField(props) {
     <div ref={fieldRef && fieldRef}>
       <div className={classnames(...classes)}>
         { label &&
+          /* eslint-disable jsx-a11y/label-has-for */
```
Comment (magcurt8) [inline src/molecules/formfields/TextAreaField/index.js]: `eslint-disable-next-line` makes it so that you don't have to use two comments and just disables the next line rather then everything following

File: src/molecules/formfields/TextField/index.js (hunk 9 above)
Comment (magcurt8) [reply src/molecules/formfields/TextField/index.js]: I don't think it's a huge deal to keep this as-is but it might be better to just remove that rule since I don't really like linters determining our code complexity for us

File: src/molecules/formfields/util/BaseDateField/index.js [hunk 25]
```diff
@@ -112,36 +112,39 @@ class BaseDateField extends React.Component {
...
+
     const dayString = String(dayValue);
-    const maxDate = get(LAST_DATE_MONTH, `${Number(this.state.monthValue)}`, 32);
+    const maxDate = get(LAST_DATE_MONTH, `${Number(monthValue)}`, 32);
 
     return !dayString || inRange(+dayValue, 0, maxDate);
   }
 
-  set month(monthValue) {
+  set month(value) {
```
Comment (magcurt8) [inline src/molecules/formfields/util/BaseDateField/index.js]: shouldn't this be `monthValue` still?

File: src/molecules/formfields/util/BaseDateField/index.js [hunk 26]
```diff
@@ -170,95 +173,92 @@ class BaseDateField extends React.Component {
...
-            value: this.state.yearValue,
-            ...input,
-          }}
-          {...props}
-        />
-      )
-    ;
+    const { yearValue } = this.state;
+
+    return ({ input = {}, ...props } = {}) => this.wrapChild(
```
Comment (magcurt8) [inline src/molecules/formfields/util/BaseDateField/index.js]: can you add brackets to keep this broken up?

File: src/molecules/lists/ArticleList/index.js (hunk 11 above)
Comment (magcurt8) [reply src/molecules/lists/ArticleList/index.js]: if it accomplishes the same thing idt we need both!

File: src/organisms/cards/PartnerCard/index.js [hunk 27]
```diff
@@ -46,37 +46,43 @@ function PartnerCard(props) {
...
   /**
    * Provides context for the center text
    */
   centerLabel: PropTypes.string,
+
   /**
    * Short and snappy center text
    */
-  centerText: PropTypes.string.isRequired
+  centerText: PropTypes.string
```
Comment (magcurt8) [inline src/organisms/cards/PartnerCard/index.js]: why remove `.isRequired`?

File: src/organisms/cards/SimpleFeaturedPolicyCard/index.js [hunk 28]
```diff
@@ -33,53 +33,56 @@ function SimpleFeaturedPolicyCard(props) {
...
-        { premium.price ?
+        { premium.price ? (
           <div className={styles['premium']}>
             <Text
               type={3}
               font='a'
             >
-              {formattedPremium} <Text tag='span' type={11} font='a' spaced color='neutral-2'>{`/${premium.format.toUpperCase()}`}</Text>
+              {formattedPremium}
+              {' '}
```
Comment (magcurt8) [inline src/organisms/cards/SimpleFeaturedPolicyCard/index.js]: how come there's an empty string element now?

File: src/organisms/cards/SimplePolicyCard/__tests__/simple_policy_card.spec.js [hunk 29]
```diff
@@ -14,19 +13,13 @@ describe('<SimplePolicyCard />', () => {
         price: 19.8,
         format: 'mo'
       },
-      onDetails: () => {},
+      onDetails: jest.fn(),
       compareCheckbox: {
-        onCompare: () => {},
+        onCompare: jest.fn(),
```
Comment (magcurt8) [inline src/organisms/cards/SimplePolicyCard/__tests__/simple_policy_card.spec.js]: you have an extra comma here that's showing up in the snapshot

File: src/organisms/forms/AddForm/index.js [hunk 30]
```diff
@@ -60,7 +62,7 @@ AddForm.propTypes = {
   /**
    * determines if the close icon is present
    */
-  closeIcon: PropTypes.bool.isRequired,
+  closeIcon: PropTypes.bool,
```
Comment (magcurt8) [inline src/organisms/forms/AddForm/index.js]: `.isRequired`? (also if the answer is the same for all of these you don't have to comment, just want to make sure none of them slip through the cracks

File: src/organisms/tables/ComparisonTable/index.js [hunk 31]
```diff
@@ -23,9 +25,7 @@ const ComparisonTable = (props) => {
   return (
     <div className={tableClass}>
       {
-        React.Children.map(children, child =>
-          React.isValidElement(child) ? React.cloneElement(child, { expandable, expanded, tableItems }) : child
-        )
+        React.Children.map(children, (child) => React.isValidElement(child) ? React.cloneElement(child, { expandable, expanded, tableItems }) : child)
```
Comment (magcurt8) [inline src/organisms/tables/ComparisonTable/index.js]: brackets can keep the multi-line arrow functions

File: src/organisms/tables/ComparisonTable/index.js [hunk 32]
```diff
@@ -111,9 +111,7 @@ class TableHeader extends React.Component {
           <Col className={styles['col-offset-header']} />
 
           {
-            this.headerToRender.map((child, idx) =>
-              <Col key={idx} className={styles['col-offset-header-cells']}>{ child }</Col>
-            )
+            this.headerToRender.map((child, idx) => <Col key={idx} className={styles['col-offset-header-cells']}>{ child }</Col>)
```
Comment (magcurt8) [inline src/organisms/tables/ComparisonTable/index.js]: same here for multi-line arrow functions, I think you can use parens actually for jsx elements

File: src/utils/elementDistanceFromTop.js [hunk 33]
```diff
@@ -1,5 +1,6 @@
 export default (element) => {
   let pos = 0;
```
Comment (magcurt8) [inline src/utils/elementDistanceFromTop.js]: again, with spaces between var instantiation

File: src/utils/fieldUtils/__tests__/renderTooltip.spec.js [hunk 34]
```diff
@@ -2,10 +2,11 @@ import { mount } from 'enzyme';
 
 import Icon from 'atoms/Icon';
 import Modal from 'molecules/Modal';
-import { renderTooltip } from '../';
+import { renderTooltip } from '..';
 
 describe('renderTooltip()', () => {
   let className;
+
```
Comment (magcurt8) [inline src/utils/fieldUtils/__tests__/renderTooltip.spec.js]: var decs

File: yarn.lock [hunk 35]
```diff
@@ -5548,36 +6717,41 @@ jest@21.0.0:
...
-  resolved "https://registry.yarnpkg.com/js-yaml/-/js-yaml-3.10.0.tgz#2e78441646bd4682e963f22b6e92823c309c62dc"
+js-yaml@^3.13.1, js-yaml@^3.4.3, js-yaml@^3.5.1, js-yaml@^3.6.1, js-yaml@^3.7.0:
+  version "3.13.1"
+  resolved "https://registry.yarnpkg.com/js-yaml/-/js-yaml-3.13.1.tgz#aff151b30bfdfa8e49e05da22e7415e9dfa37847"
+  integrity sha512-YfbcO7jXDdyj0DGxYVSlSeQNHbD7XPWvrVWeVUujrQEoZzWJIRrCPoyk6kL6IAjAG2IolMK4T0hNUe0HOUs5Jw==
   dependencies:
     argparse "^1.0.7"
     esprima "^4.0.0"
 
 js-yaml@~3.11.0:
```
Comment (magcurt8) [inline yarn.lock]: I hate to be the bearer of bad news but it looks like something is still using version 3.11.0 and 3.7.0

File: package.json [hunk 36]
```diff
@@ -66,15 +66,14 @@
...
-    "eslint-loader": "^1.7.1",
-    "eslint-plugin-import": "2.2.0",
-    "eslint-plugin-jsx-a11y": "2.2.3",
+    "eslint-loader": "^3.0.0",
+    "eslint-plugin-import": "^2.19.1",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-markdown": "https://github.com/jmcolella/eslint-plugin-markdown/tarball/master",
-    "eslint-plugin-react": "^6.8.0",
+    "eslint-plugin-react": "^7.17.0",
     "extract-text-webpack-plugin": "^2.1.0",
```
Comment (magcurt8) [inline package.json]: I think you'll want to add a resolution here for js-yaml 3.13.1 since some other dependency tree is using a couple of the older versions

File: src/atoms/Br/__tests__/br.spec.js [hunk 37]
```diff
@@ -1,17 +1,9 @@
 import React from 'react';
-import { mount } from 'enzyme';
 import renderer from 'react-test-renderer';
 
 import Br from 'atoms/Br';
 
 describe('<Br />', () => {
-
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/atoms/Br/__tests__/br.spec.js]: `jest/no-identical-title` was throwing errors because each of these files had two identical test names. discussed with john prior to making any changes - decided to get rid of the mounted component tests in each file.

File: src/atoms/CircularLoader/__tests__/circular_loader.spec.js [hunk 38]
```diff
@@ -1,16 +1,9 @@
 import React from 'react';
-import { mount } from 'enzyme';
 import renderer from 'react-test-renderer';
 
 import CircularLoader from 'atoms/CircularLoader';
 
 describe('<CircularLoader />', () => {
-  it('renders correctly', () => {
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/atoms/CircularLoader/__tests__/circular_loader.spec.js]: see above

File: src/molecules/formfields/GoogleAutoCompleteField/__tests__/index.spec.js [hunk 39]
```diff
@@ -20,6 +21,7 @@ describe('<GoogleAutoCompleteField />', () => {
       autocompleteAddressFields: jest.fn(),
     };
 
+    /* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/molecules/formfields/GoogleAutoCompleteField/__tests__/index.spec.js]: it has to be specifically disabled for these files - the data is received from google in snakecase so it can't be changed to camelcase here.

File: src/molecules/formfields/DateField/__tests__/date_field.spec.js [hunk 40]
```diff
@@ -4,6 +4,7 @@ import DateField from 'molecules/formfields/DateField';
 
 describe('<DateField />', () => {
   let props;
+
   let component;
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/molecules/formfields/DateField/__tests__/date_field.spec.js]: i've tried removing the space but the linter keeps adding it back in.

File: src/molecules/formfields/ShortDateField/__tests__/short_date_field.spec.js (hunk 21 above)
Comment (abehrman89) [reply src/molecules/formfields/ShortDateField/__tests__/short_date_field.spec.js]: see above

File: src/molecules/formfields/util/BaseDateField/index.js (hunk 25 above)
Comment (abehrman89) [reply src/molecules/formfields/util/BaseDateField/index.js]: had to make this change because of `react/destructuring-assignment` - had to pull `monthValue` off of `state` so I had to change the parameter name

File: src/organisms/cards/PartnerCard/index.js (hunk 27 above)
Comment (abehrman89) [reply src/organisms/cards/PartnerCard/index.js]: a `prop` cannot be both a `defaultProp` and be marked `isRequired` - discussed with john and decided to remove `isRequired` and keep `defaultProp` setting

File: src/organisms/forms/AddForm/index.js (hunk 30 above)
Comment (abehrman89) [reply src/organisms/forms/AddForm/index.js]: see above

File: src/utils/elementDistanceFromTop.js (hunk 33 above)
Comment (abehrman89) [reply src/utils/elementDistanceFromTop.js]: see above

File: src/utils/fieldUtils/__tests__/renderTooltip.spec.js (hunk 34 above)
Comment (abehrman89) [reply src/utils/fieldUtils/__tests__/renderTooltip.spec.js]: see above

File: src/molecules/formfields/TextAreaField/index.js (hunk 24 above)
Comment (abehrman89) [reply src/molecules/formfields/TextAreaField/index.js]: ooh nice! i changed all these comments to use `next-line`

File: src/atoms/List/__tests__/list.spec.js [hunk 41]
```diff
@@ -45,14 +44,16 @@ describe('<List />', () => {
...
-        <small>Now go and write some code! -- <em>You code monkey, you ;).</em></small>
+        <small>
+          Now go and write some code! -- <em>You code monkey, you ;).</em>
+        </small>
         <List>
-          {" I'm nested!! "}
-          <li>{"I'm wrapped in a 'li'"}</li>
-          <em>{"I'm wrapped in a 'em'"}</em>
+          {' I\'m nested!! '}
+          <li>I&apos;m wrapped in a &apos;li&apos;</li>
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/atoms/List/__tests__/list.spec.js]: updating `curly-brace` and `quotes` in pg config

File: src/molecules/StepIndicator/Readme.md [hunk 42]
```diff
@@ -8,26 +8,22 @@
             path: '/basics',
             currentStepActive: false,
             clickable: true,
-            nextStepAccessible: true
-          },
+            nextStepAccessible: true },
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/molecules/StepIndicator/Readme.md]: changed

File: src/organisms/cards/SimpleFeaturedPolicyCard/index.js (hunk 28 above)
Comment (abehrman89) [reply src/organisms/cards/SimpleFeaturedPolicyCard/index.js]: idk it's gone now

File: src/atoms/Img/index.js (hunk 3 above)
Comment (abehrman89) [reply src/atoms/Img/index.js]: file reverted back to master and complexity ignored

File: src/molecules/Filter/index.js (hunk 6 above)
Comment (abehrman89) [reply src/molecules/Filter/index.js]: discussed offline - leaving as `onBlur`

File: src/molecules/asides/ImageAside/index.js (hunk 7 above)
Comment (abehrman89) [reply src/molecules/asides/ImageAside/index.js]: discussed offline - leaving as `alt=''`

File: src/molecules/formfields/SelectField/index.js (hunk 8 above)
Comment (abehrman89) [reply src/molecules/formfields/SelectField/index.js]: file reverted back to master and complexity ignored

File: src/molecules/formfields/TextField/index.js (hunk 9 above)
Comment (abehrman89) [reply src/molecules/formfields/TextField/index.js]: file revered back to master and complexity ignored

File: src/molecules/lists/ArticleList/index.js (hunk 11 above)
Comment (abehrman89) [reply src/molecules/lists/ArticleList/index.js]: discussed offline - keeping `alt=''`

File: src/organisms/cards/CompanyCard/index.js (hunk 14 above)
Comment (abehrman89) [reply src/organisms/cards/CompanyCard/index.js]: discussed offline - keeping `alt=''`

File: src/molecules/formfields/DateField/__tests__/date_field.spec.js (hunk 40 above)
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/molecules/formfields/DateField/__tests__/date_field.spec.js]: spaces removed

File: src/molecules/formfields/ShortDateField/__tests__/short_date_field.spec.js (hunk 21 above)
Comment (abehrman89) [reply src/molecules/formfields/ShortDateField/__tests__/short_date_field.spec.js]: space removed

File: src/atoms/Icon/index.js [hunk 43]
```diff
@@ -27,7 +27,7 @@ class Icon extends React.Component {
 
   getSVG = () => {
     fetch(this.svgSrc)
-      .then(data => data.text())
+      .then((data) => data.text())
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/atoms/Icon/index.js]: discussed offline - keeping parens here

File: src/molecules/formfields/SliderField/index.js (hunk 23 above)
Comment (abehrman89) [reply src/molecules/formfields/SliderField/index.js]: changed

File: src/organisms/cards/SimplePolicyCard/__tests__/simple_policy_card.spec.js (hunk 29 above)
Comment (abehrman89) [reply src/organisms/cards/SimplePolicyCard/__tests__/simple_policy_card.spec.js]: fixed snapshot

File: src/organisms/tables/ComparisonTable/index.js (hunk 31 above)
Comment (abehrman89) [reply src/organisms/tables/ComparisonTable/index.js]: changed

File: src/organisms/tables/ComparisonTable/index.js (hunk 32 above)
Comment (abehrman89) [reply src/organisms/tables/ComparisonTable/index.js]: changed

File: yarn.lock (hunk 35 above)
Comment (abehrman89) [reply yarn.lock]: `js-yaml` resolved to `3.13.1`

File: src/molecules/formfields/SliderField/index.js [hunk 44]
```diff
@@ -28,17 +30,19 @@ class SliderField extends Component {
       sliderValues,
     } = this.props;
 
+    const { sliderIndex } = this.state;
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/molecules/formfields/SliderField/index.js]: moved

File: src/organisms/cards/SimpleFeaturedPolicyCard/index.js [hunk 45]
```diff
@@ -44,42 +44,43 @@ function SimpleFeaturedPolicyCard(props) {
...
+            )
         }
 
-
         {
-          onContinue &&
+          onContinue && (
             <React.Fragment>
-              <Spacer size={36} />
+              <Spacer size={36} />,
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/organisms/cards/SimpleFeaturedPolicyCard/index.js]: removed comma

File: src/organisms/cards/SimplePolicyCard/index.js [hunk 46]
```diff
@@ -46,23 +46,24 @@ function SimplePolicyCard(props) {
...
             {
-              !isEmpty(compareCheckbox) &&
-              <React.Fragment>
-                <Compare {...compareCheckbox} />
-                <div className={styles['divider']} />
-              </React.Fragment>
-            }
+              !isEmpty(compareCheckbox) && (
+                <React.Fragment>
+                  <Compare key={1} {...compareCheckbox} />
```
[the remark this answers is not in the document]
Comment (abehrman89) [reply src/organisms/cards/SimplePolicyCard/index.js]: Yup - this is from when the linter removed the `React.Fragment` and made it an `array` for some reason. Keys removed.
